cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.md

```diff
@@ -1,400 +1,207 @@
 
-Inside quoted event handler:
+### Output Encoding for “JavaScript Contexts”
 
-```html
-<div onmouseover="x='...ENCODE UNTRUSTED DATA BEFORE PUTTING HERE...'"</div>
-```
+“JavaScript Contexts” refer to placing variables into inline JavaScript which is then used in an HTML document. We commonly see this in WYSIWYG editors, tutorials for software development, and sandboxes for people to test software products. 
 
-<!-- textlint-disable terminology -->
-Please note there are some JavaScript functions that can never safely use untrusted data as input - **EVEN IF JAVASCRIPT ENCODED!**
-<!-- textlint-enable -->
+The only ‘safe’ location for placing variables in JavaScript is inside a “quoted data value”. All other contexts are unsafe and you should not place variable data in them.
 
-For example:
+Examples of “Quoted Data Values”
 
 ```html
-<script>
-window.setInterval('...EVEN IF YOU ENCODE UNTRUSTED DATA YOU ARE XSSED HERE...');
-</script>
+<script>alert('$varUnsafe’)</script>
+<script>x=’$varUnsafe’</script>
+<div onmouseover="'$varUnsafe'"</div>
 ```
 
-Except for alphanumeric characters, encode all characters with the `\xHH` format to prevent switching out of the data value into the script context or into another attribute. **DO NOT** use any escaping shortcuts like `\"` because the quote character may be matched by the HTML attribute parser which runs first. These escaping shortcuts are also susceptible to **escape-the-escape attacks** where the attacker sends `\"` and the vulnerable code turns that into `\\"` which enables the quote.
-
-If an event handler is properly quoted, breaking out requires the corresponding quote. However, we have intentionally made this rule quite broad because event handler attributes are often left unquoted. Unquoted attributes can be broken out of with many characters including `[space]` `%` `*` `+` `,` `-` `/` `;` `<` `=` `>` `^` and `|`.
-
-Also, a `</script>` closing tag will close a script block even though it is inside a quoted string because the HTML parser runs before the JavaScript parser. Please note this is an aggressive encoding policy that over-encodes. If there is a guarantee that proper quoting is accomplished then a much smaller character set is needed. Please look at the [OWASP Java Encoder](https://wiki.owasp.org/index.php/OWASP_Java_Encoder_Project#tab=Use_the_Java_Encoder_Project) JavaScript encoding examples for examples of proper JavaScript use that requires minimal encoding.
-
-#### RULE \#3.1 - HTML Encode JSON values in an HTML context and read the data with JSON.parse
+Encode all characters using the `\xHH` format. Do not use other formats or escape strings. Escaping, changing execution context, and abusing the parser order can all trigger XSS.
```

Comment:
Yes so the thought process for this was "I am heavily editing 40 pages of content into 10". 🤣 

So, the advice here is from the prior version of the cheatsheet and while it might still be accurate, I was planning on doing a second pass to review the currency and applicability of the technical aspects after the overall rework. Hope that clarifies why the `\xHH` recommendation is there for now.